index.js

```diff
@@ -138,6 +149,270 @@ app.intent('send.message - no', async conv => {
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  // If target isn't undefined
+  if (target) {
+    calls.forEach(c => {
+      if (c.title.toLowerCase() === target.toLowerCase()) {
+        matches.push(c);
+      }
+    });
+
+    if (!matches.length) {
+      let prompt = `<speak>I cannot find any conference call with name ${target}. Here are the names of your current ongoing conferences. 
+      Which would you like to join?</speak>`;
+      let titles = calls.map((call) => call.title.toLowerCase());
+      conv.ask(prompt, new Suggestions(titles));
+      conv.contexts.set('joinconference_gettarget', 5, {
+        calls: calls
+      });
+      return;
+    }
+
+    if (matches.length === 1) {
+      // One result found. Ask user for confirmation.
+      const callId = matches[0].callId;
+      conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+      conv.contexts.set('joinconference_send', 5, {
+        callId: callId
+      });
+      return;
+    }
+
+    // Multiple matches. Tell user that they can either re-enter a conference or just join the first match.
+    conv.ask(`More than one conference found with the name ${target}. Joining the first ${target} conference.`);
+    const callId = matches[0].callId;
+    conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+    conv.contexts.set('joinconference_send', 5, {
+      callId: callId
+    });
+    return;
+  }
+
+  let titles = [];
+  calls.forEach(call => {
+    if (!call.title) {
+      let placeholder = truncate(call.topicPlaceholder);
+      titles.push(placeholder);
+    } else {
+      titles.push(call.title);
+    }
+  });
+
+  let prompt = '<speak>Which conference would you like to join? Here are your ongoing conferences.</speak>';
+  conv.ask(prompt, new Suggestions(titles));
+  conv.contexts.set('joinconference_gettarget', 5, {
+    calls: calls, 
+    titles: titles
+  });
+  return;
+}) 
+
+
+/**
+ * join.conference - collect.target
+ */
+app.intent('join.conference - collect.target', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  const { calls, titles } = conv.contexts.input['joinconference_gettarget'].parameters;
+  const { target } = conv.parameters;
+  let callId = null;
+
+  // Lowercase titles so if the name of the conference is Test and they type in test
+  // it will still be found. That is the reason for the lowercasing the target and titles.
+  const lcTitles = titles.map(title => title.toLowerCase());
+  const index = lcTitles.indexOf(target);
+
+  if (index !== -1) {
+    callId = calls[index].callId;
+  } else {
+    for (let call of calls) {
```

Comment:
Since you aren't breaking, you can use forEach here. Or if you want to another think you can do to loop through an array and break if you find a something is to use .some. For example,
calls.some(call => {
if (call.title.toLowerCase() === target.toLowerCase()) {
callId = call.callId;
return true;
}
}):

Reply:
The above example will loop through and break as soon as an iteration returns true. This is just a trick, but you can still use forEach.